src/pocketmine/item/WritableBook.php

```diff
@@ -207,9 +215,42 @@ public function getTitle() : string{
 	 * @param string $title
 	 */
 	public function setTitle(string $title) : void{
-		$this->correctNBT();
-		$namedTag = $this->getNamedTag();
-		$namedTag->title->setValue($title);
+		$namedTag = $this->getCorrectedNamedTag();
+		if(isset($namedTag->title)){
+			$namedTag->title->setValue($title);
+		} else {
+			$namedTag->title = new StringTag("title", $title);
+		}
 		$this->setNamedTag($namedTag);
 	}
+
+	/**
+	 * @return CompoundTag
+	 */
+	public function getCorrectedNamedTag() : CompoundTag{
+		return $this->getNamedTag() ?? new CompoundTag();
+	}
+
+	public function getMaxStackSize() : int{
+		return 1;
+	}
+
+	/**
+	 * @param int         $id
+	 * @param CompoundTag $namedTag
+	 *
+	 * @return bool
+	 */
+	private function movePagesAboveDownwards(int $id, CompoundTag $namedTag) : bool{
+		if(!isset($namedTag->pages)){
+			return false;
+		}
+		foreach($namedTag->pages as $key => $page){
+			if(!is_numeric($key) or $key <= $id){
```

Comment:
`is_numeric` returns true for "floats", "negative numbers", floats and negative numbers. Since the keys are purely integers, maybe use `is_int`?

Reply:
This function doesn't work as intended yet. is_int wouldn't work, as the $key would never be an integer. Instead, always a string containing an integer, so an int typecast would be required either way.

Reply:
Ah, okay.